src/client/components/pages/parts/privs-edit-modal.js

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2023 Shivam Awasthi
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import * as bootstrap from 'react-bootstrap';
+import {PrivilegeTypeBits, getPrivilegeShieldIcon, getPrivilegeTitleFromBit} from '../../../../common/helpers/privileges-utils';
+import {faPencilAlt, faTimes} from '@fortawesome/free-solid-svg-icons';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import PrivilegeBadges from './privilege-badges';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const {Button, Form, Modal} = bootstrap;
+
+class PrivsEditModal extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			privs: props.targetUser.privs,
+			submittable: false
+		};
+
+		this.handleBitChange = this.handleBitChange.bind(this);
+		this.handleSubmit = this.handleSubmit.bind(this);
+	}
+
+	async handleSubmit() {
+		const {privs} = this.state;
+		const oldPrivs = this.props.targetUser.privs;
+		if (privs === oldPrivs) {
+			return;
+		}
+
+		const data = {
+			adminId: this.props.adminId,
+			newPrivs: privs,
+			oldPrivs,
+			targetUserId: this.props.targetUser.id
+		};
+
+		try {
+			const response = await fetch('/editor/privs/edit/handler', {
+				body: JSON.stringify(data),
+				headers: {
+					'Content-Type': 'application/json; charset=utf-8'
+				},
+				method: 'POST'
+			});
+			if (!response.ok) {
+				const {error} = await response.json();
+				throw new Error(error ?? response.statusText);
+			}
+			window.location.reload();
+		}
```

Comment:
It'll be better if you can avoid reloading the window, by getting the updated values from the API and updating the states.

Reply:
Agreed. Updating the states definitely seems to be a better way. I have made some changes. I am not getting updated values from the API because each 'Save Privilege' action makes only one change to the state: it changes the `privs` variable of the `targetUser`, so I have directly updated the value on the state object with the new object. What do you think?

Reply:
This solution works, but you should fetch the entire data from the API again so that in case any other use has changed the privileges or a new user gets added, it gets reflected on your end as well. So this function `updatePrivsOnResultsList` should pull a new copy of the data.

Reply:
Hmm, good point. I have made the changes. Now, I am using the `triggerSearch` function inside the Pager component to refetch the data using the same query whenever we are updating the privileges. For achieving this task, I am passing a trigger as a prop which helps us run this `triggerSearch` function(since this lies in a child Pager component) from the parent AdminPanelSearch component. What do you think of this approach?